File: Abhishek/Climbing Stairs.cpp
class Solution {
public:
    int memo[100]={0};
    int climbStairs(int n) {
        return climb_Stairs(0, n);
    }
    int climb_Stairs(int i, int n) {
        if (i > n) {
            return 0;
        }
        if (i == n) {
            return 1;
        }
        if (memo[i] > 0) {
            return memo[i];
        }
        memo[i] = climb_Stairs(i + 1, n) + climb_Stairs(i + 2, n);
        return memo[i];
    }
};

File: Rohit/DP/ClimbingStairs.java
class Solution {
    public int climbStairs(int n,int[] dp)
    {
         if(n==0 || n==1)
        {
            return 1;
        }
        if(dp[n]!=-1)
        {
           return dp[n]; 
        }
        int ans=0;
        if(n-1>=0)
        {
            ans+=climbStairs(n-1,dp);
        }
        if(n-2>=0)
        {
            ans+=climbStairs(n-2,dp);
        }
        dp[n]=ans;
        return ans;
    }
    
    
    
    public int climbStairs(int n) {
        if(n==0)
        {
            return 0;
        }
        int[] dp=new int[n+1];
        Arrays.fill(dp,-1);
        return climbStairs(n,dp);
       
    }
}
